src/object/mergeOptions.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Merges two option objects into a new object.
+ * - If both arguments are defined, properties
+ *   from the second object (`b`) will override those
+ *   from the first one (`a`) when keys overlap
+ * - If either argument is `undefined`, the other is returned
+ * - If both are `undefined`, the result is `undefined`
+ *
+ * @param a - The first options object, or `undefined`.
+ * @param b - The second options object, or `undefined`.
+ * @returns A new object containing merged properties, or `undefined` if both are `undefined`.
+ * @since v12.7.0
+ *
+ * @see https://radashi.js.org/reference/objects/mergeOptions
+ *
+ * @example
+ * ```ts
+ * mergeOptions({ a: 1, b: 2 }, { b: 3, c: 4 })
+ * // => { a: 1, b: 3, c: 4 }
+ * ```
+ *
+ * @example
+ * ```ts
+ * mergeOptions(undefined, { a: 1 })
+ * // => { a: 1 }
+ * ```
+ *
+ * @example
+ * ```ts
+ * mergeOptions({ a: 1 }, undefined)
+ * // => { a: 1 }
+ * ```
+ *
+ * @example
+ * ```ts
+ * mergeOptions(undefined, undefined)
+ * // => undefined
+ * ```
```

Comment:
Let's merge this into a single `@example` code block, since that's how we do it elsewhere.